### PR TITLE
Add Style#removeTerrain and Style#removeLight methods

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add Style#removeTerrain and Style#removeLight methods.</changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### Summary of changes

Introduce `Style.removeStyleTerrain()` and `Style.removeStyleLight()` methods that clear terrain and light sources if they were present and do nothing otherwise.

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/Style.kt [hunk 1]
```diff
@@ -324,6 +324,15 @@ class Style internal constructor(
       this.setStyleLightProperty(id, light)
     }
 
+  /**
+   * Removes light from style if it was set.
+   */
+  fun removeStyleLight() {
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/Style.kt:330]:
```suggestion
  fun removeLight() {
```

File: sdk/src/main/java/com/mapbox/maps/Style.kt [hunk 2]
```diff
@@ -359,6 +368,15 @@ class Style internal constructor(
       this.setStyleTerrainProperty(property, value)
     }
 
+  /**
+   * Removes terrain from style if it was set.
+   */
+  fun removeStyleTerrain() {
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/Style.kt:374]:
```suggestion
  fun removeTerrain() {
```

File: sdk/src/main/java/com/mapbox/maps/Style.kt (hunk 1 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/Style.kt:330]: we have `setStyleLight` method so I've selected `removeStyleLight` to be consistent with setter, @mapbox/maps-android WDYT?

Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/Style.kt:330]: setStyle* is the naming for the low level style API, in high level APIs we remove the `Style` from the function name.

And just double checked, we should add the removeLight and removeTerrain API as extension functions to the Light and Terrian class, as the higher level add APIs are extended that way as well.

**Review by ank27 — APPROVED** [review]

File: sdk/src/test/java/com/mapbox/maps/StyleTest.kt [hunk 3]
```diff
@@ -348,4 +354,10 @@ class StyleTest {
     style.setStyleTerrainProperty("id", value)
     verify { nativeMap.setStyleTerrainProperty("id", value) }
   }
+
+  @Test
+  fun removeStyleTerrain() {
+    style.removeStyleTerrain()
```
Comment (alexander-kulikovskii) [inline sdk/src/test/java/com/mapbox/maps/StyleTest.kt:360]: can I as a user call it twice?
what will happen? or it's not our responsibility

Comment (kiryldz) [reply sdk/src/test/java/com/mapbox/maps/StyleTest.kt]: it's not our responsibility.

File: sdk/src/main/java/com/mapbox/maps/Style.kt (hunk 1 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/Style.kt]: @pengdev thanks, fixed!

**Review by tobrun — APPROVED** [review]
LGTM, did you also update the templates? (we are touching generated code)

**Review by pengdev — APPROVED** [review]
LGTM

**Comment by pengdev** [comment]
Please update the title of the PR

File: extension-style/src/main/java/com/mapbox/maps/extension/style/light/generated/LightExt.kt [hunk 4]
```diff
@@ -24,4 +25,11 @@ fun StyleInterface.setLight(light: StyleContract.StyleLightExtension) {
   light.bindTo(this)
 }
 
+/**
+ * Removes light from style if it was set.
+ */
+fun StyleInterface.removeLight() {
+  setStyleLight(Value.nullValue())
+}
```
Comment (macdrevx) [inline extension-style/src/main/java/com/mapbox/maps/extension/style/light/generated/LightExt.kt:33]: @kiryldz did you verify that this works? I'm getting an error with "light must be an object"

Comment (macdrevx) [reply extension-style/src/main/java/com/mapbox/maps/extension/style/light/generated/LightExt.kt:33]: Terrain seems fine though

**Comment by kiryldz** [comment]
Noting that API to remove light was removed in https://github.com/mapbox/mapbox-maps-android/pull/974

**Pull request merged**